jenkins/JenkinsFile-rebuild-docker-images

```diff
@@ -0,0 +1,318 @@
+            ws(per_exec_ws("tvm/docker-ci-wasm")) {
+              cleanWs();
+              init_git()
+              dir('tvm'){
+                sh "${docker_build} ci_wasm --tag ${TVM_DOCKER_TAG}"
+              }
+
+              upload_to_docker_hub("ci_wasm");
+
+            }
+          } // steps
+          post {
+            always {
+              cleanup_docker_image("${BUILD_TAG}.ci_wasm:${TVM_DOCKER_TAG}");
+              cleanWs();
+            }
+          } // post
+        } // stage
+
+        stage('ci-i386') {
+          agent { node { label 'CPU' } }
+          steps {
+            ws(per_exec_ws("tvm/docker-ci-i386")) {
+              cleanWs();
+              init_git()
+              dir('tvm'){
+                sh "${docker_build} ci_i386 --tag ${TVM_DOCKER_TAG}"
+              }
+
+              upload_to_docker_hub("ci_i386");
+
+            }
+          } // steps
+          post {
+            always {
+              cleanup_docker_image("${BUILD_TAG}.ci_i386:${TVM_DOCKER_TAG}");
+              cleanWs();
+            }
+          } // post
+        } // stage
+
+        stage('ci-arm') {
+          agent { node { label 'ARM' } }
+          steps {
+            ws(per_exec_ws("tvm/docker-ci-arm")) {
+              cleanWs();
+              init_git()
+              dir('tvm'){
+                sh "${docker_build} ci_arm --tag ${TVM_DOCKER_TAG}"
+              }
+
+              upload_to_docker_hub("ci_arm");
+
+            }
+          } // steps
+          post {
+            always {
+              cleanup_docker_image("${BUILD_TAG}.ci_arm:${TVM_DOCKER_TAG}");
+              cleanWs();
+            }
+          } // post
+        } // stage
+
+        stage('ci-qemu') {
+          agent { node { label 'CPU' } }
+          steps {
+            ws(per_exec_ws("tvm/docker-ci-qemu")) {
+              cleanWs();
+              init_git()
+              dir('tvm'){
+                sh "${docker_build} ci_qemu --tag ${TVM_DOCKER_TAG}"
+              }
+
+              upload_to_docker_hub("ci_qemu");
+
+            }
+          } // steps
+          post {
+            always {
+              cleanup_docker_image("${BUILD_TAG}.ci_qemu:${TVM_DOCKER_TAG}");
+              cleanWs();
+            }
+          } // post
+        } // stage
+
+      } // parallel
+    } // stage: Build
+  } // stages
+  post {
+    success {
+      discordSend description: "New images published on DockerHub with tag `${TVM_DOCKER_TAG}`. Use `docker pull tlcpackstaging/<image_type>:${TVM_DOCKER_TAG}` to download the images. Image types: `ci_arm`, `ci_cpu`, `ci_gpu`, `ci_i386`, `ci_lint`, `ci_qemu`, `ci_wasm`.",
+        link: "https://hub.docker.com/u/tlcpackstaging/",
+        result: currentBuild.currentResult,
+        title: "${JOB_NAME}",
+        webhookURL: "${DISCORD_WEBHOOK}"
+    }
+    unsuccessful {
+      discordSend description: "Failed to generate Docker images using TVM hash `${TVM_GIT_REPO_URL}`. See logs at ${BUILD_URL}.",
```

Comment:
what if some of them succeed?

Reply:
If some of them success (this happens sometimes) we consider the build as failed, because one of the goals is to have a full set of images that represent a give hash on GitHub.

As the builds run in parallel, some of them might upload images to DockerHub as a side effect, and we can use if we want to, but we won't be doing further testing on partial builds.